web-server/v0.4/src/routes/Dashboard/Summary.js

```diff
@@ -18,6 +18,34 @@ const tabList = [
     key: 'tools',
     tab: 'Tools & Parameters',
   },
+  {
+    key: 'toc',
+    tab: 'Table of Content',
+  },
+];
+
+const DirectoryTree = Tree.DirectoryTree;
+const { TreeNode } = Tree;
+let fileData = [];
```

Comment:
`fileData` is only used within the scope of the `insertTreeData` function. Can we move this reference to that function rather than having global scope over the component?

Reply:
@gurbirkalsi the` insertTreeData` function is a recursive function so if we shift the variable inside `insertTreeData` function, we won't be able to save the previous values.

Reply:
We should be able to pass `fileData` as a parameter of the recursive function in order to keep track of the data (provided no data loss occurs) at each recursive call or track it as a component variable within the props constructor: 

```
constructor(props) {
    super(props);
    
    this.fileData = [];
}